editor/plugins/script_text_editor.cpp

```diff
@@ -638,13 +639,38 @@ bool ScriptTextEditor::is_unsaved() {
 }
 
 Variant ScriptTextEditor::get_edit_state() {
-	return code_editor->get_edit_state();
+	Dictionary state = code_editor->get_edit_state();
+
+	Array breakpoints;
+	for (const int &line : code_editor->get_text_editor()->get_breakpointed_lines()) {
+		const Breakpoint &bp = EditorDebuggerNode::get_singleton()->get_breakpoint(script->get_path(), line + 1);
+		breakpoints.append(bp.serialize());
+	}
+	state["breakpoints"] = breakpoints;
+
+	return state;
 }
 
 void ScriptTextEditor::set_edit_state(const Variant &p_state) {
 	code_editor->set_edit_state(p_state);
 
 	Dictionary state = p_state;
+
+	if (state.has("breakpoints")) {
+		Array breakpoints = state["breakpoints"];
+		for (const Variant &E : breakpoints) {
+			if (E.get_type() != Variant::DICTIONARY) {
+				// Old breakpoint data which contain only the line number.
+				code_editor->get_text_editor()->set_line_as_breakpoint(E, true);
+				continue;
+			}
+			const Dictionary &dict = E;
+			const int &line = ((int)dict["line"]);
+			code_editor->get_text_editor()->set_line_as_breakpoint(line - 1, true, dict["enabled"], dict["suspend"], dict["condition"], dict["print"]);
+			EditorDebuggerNode::get_singleton()->set_breakpoint(script->get_path(), line, true, dict["enabled"], dict["suspend"], dict["condition"], dict["print"]);
```

Comment:
Doesn't `set_line_as_breakpoint` emit a signal that connects the breakpoint to the debugger? Is this second line needed?